File: add_product.py
import mysql.connector
from flask import Flask, request, jsonify
from flask_cors import CORS, cross_origin 

app = Flask(__name__)
CORS(app, support_credentials=True)

mydb = mysql.connector.connect(
  host="localhost",
  user="root",
  password="",
  database="produits"
)  

mycursor = mydb.cursor(buffered=True)

@app.route("/produits", methods=["GET"])
@cross_origin(supports_credentials=True)
def all_produits():
    produits=None
    if request.method=='GET':
        sql=("SELECT * FROM produits.produits")
        mycursor.execute(sql)
        produits = [
            dict(id_product_google=row[0], price=row[1], date_creation=row[2], 
                 date_update=row[3],
                 id_product=row[4], reference=row[5])
            for row in mycursor.fetchall()
        ]
        if produits is not None:
            return jsonify(produits)
    
    
@app.route("/produit/<reference>", methods=["GET", "DELETE"])
@cross_origin(supports_credentials=True)
def single_product(reference):
    
    produit = None
    if request.method == "GET":
        sql=("SELECT * FROM produits.produits WHERE reference=%s")
        var=(reference,)
        print(reference)
        print(var)
        mycursor.execute(sql,var)
        produit = [
            dict(id_product_google=row[0], price=row[1], date_creation=row[2], 
                 date_update=row[3],
                 id_product=row[4], reference=row[5])
            for row in mycursor.fetchall()
        ]
        if produit is not None:
            return jsonify(produit), 200
        else:
            return "ça va pas", 404

    if request.method=="DELETE":
        mycursor.execute("DELETE * FROM produits WHERE reference=%s", (reference,))

    mydb.commit()
    
    
@app.route("/produit/<reference>/price/<new_price>", methods=["GET"])
@cross_origin(supports_credentials=True)
def majprix(reference,new_price):
    mycursor1 = mydb.cursor(buffered=True)
    print(reference, new_price)
    if request.method == "GET":
        sql1=("SELECT price FROM produits WHERE reference=%s" 
              "AND date_update=(SELECT max(date_update) FROM produits)")
        var=(reference,)
        var_suite=(reference, reference)
        sql_id=("SELECT id_product FROM produits WHERE reference=%s"
                "AND numero=(SELECT max(numero)FROM produits WHERE reference=%s)")
        id_product=mycursor1.execute(sql_id,var_suite)
        sql_id_product_google=("SELECT id_product_google FROM produits WHERE reference=%s"
                              "AND numero=(SELECT max(numero)FROM produits WHERE reference=%s)")
        id_product_google=mycursor1.execute(sql_id_product_google,var_suite) 
        sql_date_base=("SELECT date_creation FROM produits WHERE reference=%s "
                       "AND numero=(SELECT max(numero)FROM produits WHERE reference=%s)")
        date_base=mycursor1.execute(sql_date_base,var_suite)
        res1=mycursor1.execute(sql1,var)
        if res1!=new_price:
            sql = ("INSERT INTO produits (id_product_google, id_product, price, reference, date_creation)"
                   " VALUES (%s, %s, %s, %s, %s)")
            val2=(id_product_google, id_product, new_price, reference, date_base)
            mycursor1.execute(sql,val2)
            mydb.commit()
        
        return " c bn",201
    
    #http://127.0.0.1:9090/majprix/2/17
    
@app.route("/histoprix/<reference>", methods=["GET"])
@cross_origin(supports_credentials=True)
def histoprix(reference):
    produits=None
    if request.method=="GET":
        sql=("SELECT price,date_update FROM produits WHERE reference = %s")
        var=(reference,)
        mycursor.execute(sql,var)
        produits = [
            dict(price=row[0], date_update=row[1])
            for row in mycursor.fetchall()
        ]
        if produits is not None:
            return jsonify(produits)


    
    
if __name__ == "__main__":
    app.run(debug=True, port=9090)

File: addproduct.php

<?php
try{
	$connect=new PDO("mysql:host=mysql.hostinger.fr;dbname=u604012333_lboun;charset=utf8","u604012333_lboun","56umQEklxoA8MUg5E1");


	$id_product_google = isset($_GET['id_product_google']) ? $_GET['id_product_google']:null;
	$price = isset($_GET['price']) ? $_GET['price']:null;
	$date_creation= date("Y-m-d H:i:s");
	$date_update = 	date("Y-m-d H:i:s");
	$id_product	= isset($_GET['id_product']) ? $_GET['id_product']:null;
	$reference	= isset($_GET['reference']) ? $_GET['reference']:null;
	
	$lastID_n_produit = $connect->query("SELECT id_product FROM product where id_product = '".$id_product."'");
	$rowID_n_produit=$lastID_n_produit->fetch();
	//$diff = date_diff()
	
	    $reponse = $connect->exec("INSERT INTO product(id_product_google,price,date_creation,date_update,id_product,reference) values ('".$id_product_google."','".$price."','".$date_creation."','".$date_update."','".$id_product."','".$reference."')");    
	//create table productUnique(id_product_google varchar(100) primary key)
	$reponse1 = $connect->exec("INSERT INTO productUnique(id_product_google,reference) values ('".$id_product_google."','".$reference."') ");
	

	//$connect->commit();
}catch(Exception $ex){
	echo ": ".$ex;
	//$connect->rollBack();
	
}
?>